src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,105 @@
+package duke;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.storage.Storage;
+import duke.ui.Ui;
+import duke.parser.Parser;
+import duke.data.task.Task;
+import duke.data.TaskList;
+import duke.data.exception.DukeException;
+public class Duke {
+
+    private final Storage storage;
+    private TaskList tasklist;
+    private final Ui ui;
+    private final Parser parser;
+
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        parser = new Parser();
+        try {
+            tasklist = new TaskList(storage.loadTasks());
+        } catch (DukeException e) {
+            tasklist = new TaskList();
+        }
+    }
+
+    public void run() {
+        this.ui.greet();
+        Scanner scanner = new Scanner(System.in);
+
+        while (true) {
+            String input = scanner.nextLine();
+            String keyword = parser.getInstruction(input);
+            try {
+                switch (keyword) {
+                case "bye": {
+                    this.ui.exitBot();
+                    return;
+                }
+                case "list": {
+                    this.ui.printList(tasklist.getTaskList());
```

Comment:
As other lines you use this.tasklist, maybe here should be same?
```suggestion
                    this.ui.printList(this.tasklist.getTaskList());
```

Reply:
Indeed, sometimes `this.` is used while other times it is not (for example, in the `Duke` constructor above). Perhaps something to consider.